web/elm/src/Tooltip.elm

```diff
@@ -1,57 +1,135 @@
-module Tooltip exposing (Model, handleCallback, view)
-
-import Build.Styles
+module Tooltip exposing
+    ( Alignment(..)
+    , Direction(..)
+    , Model
+    , Tooltip
+    , handleCallback
+    , view
+    )
+
+import Browser.Dom
 import EffectTransformer exposing (ET)
 import HoverState exposing (TooltipPosition(..))
 import Html exposing (Html)
-import Message.Callback exposing (Callback(..), TooltipPolicy(..))
+import Html.Attributes exposing (style)
+import Message.Callback exposing (Callback(..))
 import Message.Effects as Effects
-import Message.Message as Message
-import SideBar.Styles
+import Message.Message exposing (DomID(..), Message)
 
 
 type alias Model m =
     { m | hovered : HoverState.HoverState }
 
 
+type alias Tooltip =
+    { body : Html Message
+    , arrow : Maybe Arrow
+    , attachPosition : AttachPosition
+    }
+
+
```

Comment:
Sorry I'm nagging here, can we add a short comment to explain what `Arrow` means here? Just so that when ppl read it, they can understand it refers to the pointer that points to the tooltip? =)